### PR TITLE
Add http problem container class

### PR DESCRIPTION
**Pull request by drsnyder**
Add a container class for [http problem](https://tools.ietf.org/html/draft-nottingham-http-problem-07#page-3). This will allow us to provide predictable [error responses that comply with the guidelines](https://github.com/Wikia/guidelines/tree/master/APIDesign#error-handling).

To use the class you can do the following in your resource:

```java
      Problem response =
          Problem.builder()
              .status(Response.Status.NOT_FOUND.getStatusCode())
              .type(new URI("/user-not-found.html"))
              .title(String.format("No preferences found for user %d", userId))
              .build();

      return
          Response.status(Response.Status.NOT_FOUND)
              .entity(response)
              .type(Problem.MEDIA_TYPE)
              .build();
```

https://wikia-inc.atlassian.net/browse/SERVICES-17

/cc @nmonterroso @jogura

### CODE REVIEW

**Comment by nmonterroso** [comment]
LGTM

File: gradle/pandora_dependencies.gradle [hunk 1]
```diff
@@ -11,6 +11,8 @@ dependencies {
 
     compile 'com.wikia:dropwizard-logstash-encoder:1.0.2'
 
+    compile "org.projectlombok:lombok:1.12.6"
```
Comment (jogura) [inline gradle/pandora_dependencies.gradle]: Why are we using this version of lombok? And not `1.16.2` from [Maven Central](http://search.maven.org/#search%7Cga%7C1%7Clombok)?

File: lib/service-core/src/main/java/com/wikia/pandora/http/Problem.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package com.wikia.pandora.http;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Value;
+import lombok.experimental.Builder;
```
Comment (jogura) [inline lib/service-core/src/main/java/com/wikia/pandora/http/Problem.java]: This reference is deprecated the correct one to use is `lombok.Builder`

File: gradle/pandora_dependencies.gradle (hunk 1 above)
Comment (drsnyder) [inline gradle/pandora_dependencies.gradle]: I think it was a typo. I switched the last 2 numbers in the version :(. This probably explains the comment below as well. I'm guessing it probably didn't exist in this version.

Comment (drsnyder) [inline gradle/pandora_dependencies.gradle]: Fixed in 70f5946.

**Pull request merged**